### PR TITLE
Make sum/prod/mean follow the same pattern other reductions do

### PR DESCRIPTION
**Pull request by ev-br**
goes on top of gh-17

### CODE REVIEW

File: torch_np/_ndarray.py [hunk 1]
```diff
@@ -436,19 +439,57 @@ def sum(self, axis=None, dtype=None, out=None, keepdims=NoValue,
...
+
+        axis = _helpers.allow_only_single_axis(axis)
+
+        torch_dtype = _helpers.float_or_default(dtype, self.dtype)
+        if axis is None:
+            result = self._tensor.prod(dtype=torch_dtype)
+        else:
+            result = self._tensor.prod(dtype=torch_dtype, dim=axis)
+
+        return result
```
Comment (lezcano) [inline torch_np/_ndarray.py:464]: Nit. You can golf these sort of patterns as
```suggestion
kwargs = {"dim": axis} if axis is not None else {}
return self._tensor.prod(dtype=torch_dtype, **kwargs)
```

File: torch_np/_ndarray.py [hunk 2]
```diff
@@ -436,19 +439,57 @@ def sum(self, axis=None, dtype=None, out=None, keepdims=NoValue,
...
+
+
+    @axis_out_keepdims_wrapper
+    def std(self, axis=None, dtype=None, out=None, ddof=0, keepdims=NoValue, *,
+            where=NoValue):
+        if where is not None:
+            raise NotImplementedError
+
+        torch_dtype = _helpers.float_or_default(dtype, self.dtype, enforce_float=True)
+        tensor = self._tensor.to(torch_dtype)   # XXX: needed?
```
Comment (lezcano) [inline torch_np/_ndarray.py:474]: Yeah, I think so.

**Review by lezcano — APPROVED** [review]
LGTM

**Comment by ev-br** [comment]
Thanks, merged straight to main.

**Pull request closed without merging**